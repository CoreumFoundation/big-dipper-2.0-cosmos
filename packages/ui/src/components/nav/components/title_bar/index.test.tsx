import renderer, { act } from 'react-test-renderer';
import TitleBar from '@/components/nav/components/title_bar';
import MockTheme from '@/tests/mocks/MockTheme';
import axios from 'axios';

// ==================================
// global setup
// ==================================
let component: renderer.ReactTestRenderer;

// ==================================
// mocks
// ==================================

const mockUseNavContext = {
  title: undefined as string | undefined,
  price: 0,
  marketCap: 0,
  inflation: 0,
  communityPool: 0,
};

// ==================================
// unit tests
// ==================================
describe('screen: Nav/TitleBar', () => {
  beforeEach(() => {
    component = renderer.create(
      <MockTheme>
        <TitleBar />
      </MockTheme>
    );
  });

  const mAxiosResponse = {
    data: { coreum: { usd: 1 } },
  };

  jest.spyOn(axios, 'get').mockResolvedValueOnce(mAxiosResponse);

  it('it renders', () => {
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  act(() => {
    it('hook toggles correctly', () => {
      mockUseNavContext.title = 'Validators';
      component.update(
        <MockTheme>
          <TitleBar />
        </MockTheme>
      );
      const tree = component?.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

afterEach(() => {
  mockUseNavContext.title = undefined;
  jest.clearAllMocks();
});
