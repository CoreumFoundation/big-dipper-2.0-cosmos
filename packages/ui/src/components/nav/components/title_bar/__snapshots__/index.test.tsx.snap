// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Nav/TitleBar hook toggles correctly 1`] = `
.emotion-0 {
  padding: 8px 16px;
  display: grid;
}

.emotion-1 {
  width: 100%;
  background: #FFFFFF;
  margin-top: 16px;
  border-radius: 4px;
  padding: 8px;
  display: grid;
  place-items: center start;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  width: 100%;
}

.emotion-2 .label {
  color: #414141;
  margin-right: 8px;
}

.emotion-2 .data {
  text-align: right;
}

@media (min-width:375px) {
  .emotion-2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

@media (min-width:1280px) {
  .emotion-2 {
    padding: 0;
  }
}

.emotion-3 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Noto Sans",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        marketCap
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        $0
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        inflation
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0%
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        apr
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0%
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        supply
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0 
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        communityPool
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0 
      </p>
    </div>
  </div>
</div>
`;

exports[`screen: Nav/TitleBar it renders 1`] = `
.emotion-0 {
  padding: 8px 16px;
  display: grid;
}

.emotion-1 {
  width: 100%;
  background: #FFFFFF;
  margin-top: 16px;
  border-radius: 4px;
  padding: 8px;
  display: grid;
  place-items: center start;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  width: 100%;
}

.emotion-2 .label {
  color: #414141;
  margin-right: 8px;
}

.emotion-2 .data {
  text-align: right;
}

@media (min-width:375px) {
  .emotion-2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

@media (min-width:1280px) {
  .emotion-2 {
    padding: 0;
  }
}

.emotion-3 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Noto Sans",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        marketCap
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        $0
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        inflation
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0%
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        apr
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0%
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        supply
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0 
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 label emotion-3"
      >
        communityPool
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 data emotion-3"
      >
        0 
      </p>
    </div>
  </div>
</div>
`;
